### PR TITLE
NAS-114214 / 22.02 / NAS-114214: Deleting a dataset causes the deletion of a replication task

### PR DESCRIPTION
**Pull request by bvasilenko**
Testing:
1. Storage -> Create 2 Datasets
2. Create ZVol inside each dataset
3. Data Protection -> Replication task ZVol -> ZVol2. You may also try ZVol -> Dataset, etc.
4. Storage -> Delete ZVol2
Expected result: confirmation dialog will have a list of related replication tasks, which will be deleted automatically

### CODE REVIEW

File: src/app/pages/storage/volumes/volumes-list/volumes-list-table-config.ts [hunk 1]
```diff
@@ -1105,6 +997,186 @@ export class VolumesListTableConfig implements EntityTableConfig {
...
+
+    return res;
+  }
+
+  private getDatasetRelatedSummary(
+    datasetName: string,
+    attachments: PoolAttachment[],
+    processes: PoolProcess[],
+  ): string {
+    let res = '';
```
Comment (RehanY147) [inline src/app/pages/storage/volumes/volumes-list/volumes-list-table-config.ts:1053]: Here too

File: src/app/pages/storage/volumes/volumes-list/volumes-list-table-config.ts [hunk 2]
```diff
@@ -1105,6 +997,186 @@ export class VolumesListTableConfig implements EntityTableConfig {
     return actions as EntityTableAction[];
   }
 
+  private getStorageFlattenedAttachments(attachments: PoolAttachment[]): string {
+    const flattenAttachment = (obj: PoolAttachment): string => obj.attachments.map((a) => a.split(',').map((i) => `<br> - ${i}`).join()).join();
```
Comment (RehanY147) [inline src/app/pages/storage/volumes/volumes-list/volumes-list-table-config.ts:1001]: We should break down this code into multiple properly named variable assignments on multiple lines so its more readable and understandable what's happening.

**Review by RehanY147 — CHANGES_REQUESTED** [review]
The warning shows up if I try to delete the source dataset of the replication task. But deleting the destination dataset/zvol of the replication task shows no warning about replication task. Not sure if that's intended.

**Comment by bvasilenko** [comment]
>Not sure if that's intended. 
To me that seems to be by design. Because if the destination Dataset/ZVol does not exists, it will be created automatically.

**Review by RehanY147 — APPROVED** [review]
Looks good to me

**Pull request merged**